src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs

```diff
@@ -804,14 +821,17 @@ public bool RemoveTransientTransactions(uint256 depositId = null)
         /// <inheritdoc />
         public IEnumerable<IWithdrawal> GetWithdrawals()
         {
-            foreach (TransactionData transactionData in this.Wallet.MultiSigAddress.Transactions.OrderByDescending(t => t.CreationTime))
+            lock (this.lockObject)
             {
-                Transaction walletTrx = transactionData.GetFullTransaction(this.network);
-                IWithdrawal withdrawal = this.withdrawalExtractor.ExtractWithdrawalFromTransaction(walletTrx, transactionData.BlockHash, transactionData.BlockHeight ?? 0);
-                if (withdrawal == null)
-                    continue;
+                foreach (TransactionData transactionData in this.Wallet.MultiSigAddress.Transactions.OrderByDescending(t => t.CreationTime))
```

Comment:
If you ToList() `this.Wallet.MultiSigAddress.Transactions` it will achieve the same thing as the lock.

Reply:
This  method is being removed in #3582.